docs2/site/docs/analyzers/GQL011_MustNotBeConvertibleToGraphType.md

```diff
@@ -0,0 +1,111 @@
+# GQL011: The type must not be convertible to `IGraphType`
+
+|                        | Value  |
+| ---------------------- | ------ |
+| **Rule ID**            | GQL011 |
+| **Category**           | Usage  |
+| **Default severity**   | Error  |
+| **Enabled by default** | Yes    |
+| **Code fix provided**  | No     |
+
+## Cause
+
+This rule triggers when a type implementing `IGraphType` used as the type
+argument when the type argument must NOT be convertible to `IGraphType`.
+
+## Rule description
+
+Multiple generic types and methods in the GraphQL library receive unconstrained
+type arguments. This diagnostic rule simulates `where T : not IGraphType`
+constraint which is not supported in .NET.
+
+## How to fix violations
+
+Use appropriate type argument.
+
+## Example of a violation
+
+In the following example the `PersonGraphType` is incorrectly used as the type
+argument of the `InterfaceGraphType<TSource>` type.
+
+```c#
+public class MyInterfaceGraphType : InterfaceGraphType<PersonGraphType>
+{
+}
+
+public class PersonGraphType : ObjectGraphType<Person>
+{
+}
+
+public class Person { }
+```
+
+In this example, the `StringGraphType` is incorrectly used as the return type of
+the resolver
+
+```c#
+public class MyGraphType : ObjectGraphType
+{
+    public MyGraphType()
+    {
+        Field<StringGraphType>("name")
+            .Returns<string>()
```

Comment:
Is this sample correct? (Which should be a sample of incorrect code)

Reply:
Yep, I put here the correct sample, and in the fix section, the incorrect...